branches-in-git/problem-set-git.md

```diff
@@ -0,0 +1,244 @@
+
+As usual, once we finish a feature branch, we'll want to merge it back into `main`.
+
+- [ ] Switch back to the `main` branch.
+
+- [ ] Merge TODO into `main` with `$ git merge TODO`.
+
+## Part 3: Merging Overlapping Changes
+
+We're assigned yet another feature to build:
+
+TODO
+
+For this part, we will follow some directions to force us to resolve a merge conflict.
+
+### Create a Feature Branch
+
+- [ ] Create a feature branch named TODO.
+
+### Add Simon's Work to `main`
+
+For this step, let's pretend that we're a separate developer named Simon.
+
+Simon is going to push a commit onto the `main` branch.
+
+- [ ] Switch to the `main` branch
+
+- [ ] Add this content to TODO:
+
+TODO
+
+- [ ] Make a commit with this work with the message, "I'm Simon and I'm adding TODO"
+
+### Switch to the Feature Branch
+
+Return to being yourself, who was in the middle of development on a feature branch!
+
+- [ ] Switch to the TODO branch
+
+### Build Up the Feature Branch With Overlap
+
+- [ ] Add this content to TODO:
+
+TODO
+
+- [ ] Make a commit with the message, "TODO."
+
+### Merge `main` Into the Feature Branch
+
+We want Simon's work on `main` in our current feature branch.
+
+- [ ] Begin to merge with `$ git merge main`
+
+### Resolve Conflicts
+
+- [ ] Notice the output that Git prints to the terminal. Confirm that it says that a merge conflict came up during the process of merging.
+
+- [ ] Run `$ git status` in order to see the message once again.
+
+- [ ] Notice that Git suggests resolving the conflict by adding resolved files to staging.
+
+- [ ] Notice that Git suggests actions to do with the merge, such as running `git merge --continue` or `git merge --abort`.
+
+- [ ] Open the project in VS Code or a similar text editor.
+
+- [ ] Find the location of the conflicting code. Change the lines of code to this compromise of code:
+
+TODO
+
+### !callout-info
+
+## Deciding How to Resolve Conflicts
+
+How did we decide what the code should look like at this point? We needed to use context clues and the situation at hand to decide what lines of code to keep, remove, or modify.
+
+### !end-callout
+
+- [ ] Return to the terminal and run `$ git status` again.
+
+- [ ] Add the TODO file to staging.
+
+- [ ] Make a commit **with no commit message** using `$ git commit`.
+
+- [ ] Accept the default commit message for this merge commit.
+
+- [ ] Run `$ git status` to confirm that the merge has concluded.
+
+- [ ] Run `$ git log` to check the current git history. Check that Simon's commits, your commits, and the merge commit are all present.
+
+### !callout-success
+
+## Congratulations!
+
+You just completed Parts 1-3 of this problem set!
+
+### !end-callout
+
+## Part 4
+
+Work through the following sections of [https://learngitbranching.js.org/](https://learngitbranching.js.org/):
+
+- Introduction Sequence
+  - All lessons
+- Ramping Up
+  - All lessons
+- Moving Work Around
+  - Lesson 1: Cherry-pick Intro
```

Comment:
I made these changes in the update branch as well, but for visibility I wanted to mention them here.

I strongly suggest leaving off lesson 4 in the intro (rebase) and the cherry picking stuff. These are both topics that tend to be confusing, and I don't think we should be encouraging students to use rebase at all. In my mind, the less that is planted as a valid strategy, the better. As for cherry picking, I have never needed to do it in my entire career, and for the level exposure of the students, it seems more likely to confuse than illuminate.

```suggestion
- Introduction Sequence
  - Lessons 1, 2, and 3
- Ramping Up
  - All lessons
```